### PR TITLE
VB-811: Option to choose open/closed visit type for prisoners with 'closed' restriction

### PR DESCRIPTION
**Pull request by tpmcgowan**
If the **prisoner** has a closed restriction then, after selecting visitors, show a page to choose visit type (open/closed) before proceeding to choose visit date/time. If a closed visit is chosen the reason for this ('prisoner') is stored to display in the subsequent banner.

If the **visitor** has a closed restriction then the visit type page is not used and the visit is automatically set to CLOSED.

### CODE REVIEW

File: server/routes/bookAVisit.ts
Comment (psoleckimoj) [inline server/routes/bookAVisit.ts:130]: Might be clearer to use a `.some` rather than `.reduce`? Would make it clearer what it's doing?

File: server/routes/bookAVisit.ts
Comment (psoleckimoj) [inline server/routes/bookAVisit.ts:134]: Could combine these into a ternary but that's up to you :) Means it's a single return statement.

**Review by psoleckimoj — DISMISSED** [review]
LGTM, up to you if you want to make those changes but happy to merge as is.

File: server/routes/bookAVisit.ts
Comment (tpmcgowan) [reply server/routes/bookAVisit.ts]: Good spot! Have updated to make it simpler.

File: server/routes/bookAVisit.ts
Comment (psoleckimoj) [inline server/routes/bookAVisit.ts:132]: Can actually do on 1 (2 after prettier) lines but I'm going to approve and merge anyways 🙂 

`const closedVisitPrisoner = visitSessionData.prisoner.restrictions.some(restriction => restriction.restrictionType === 'CLOSED')`

Comment (tpmcgowan) [reply server/routes/bookAVisit.ts:132]: I should have spotted that as I've got something similar a couple of lines above 🤦‍♂️!

**Pull request merged**
